### DISCUSSION TITLE
Update query with regular get request?

### DISCUSSION THREAD

**Discussion by fabianwohlfart** [bigskysoftware/htmx#1850] [Q&A]
Hi,

I am new to HTMX and are a bit stuck.

Basically I have a form, which should update something on change.
The data comes from an api:

```
<form
  id="filter"
  hx-get="/api/projects"
  hx-trigger="change"
  hx-target="#projects"
  hx-push-url="/projects"
  hx-include="#search"
  >
  <input type="radio" name="status" value="draft" checked>
  <input type="radio" name="status" value="listed">
</form>
```

I would expect the url to update with query parameters from the form, but it does not?
The get request itself is correct, e.g. /api/projects?status=draft.
But the window location won't get updated?

On the other side.
If I do this

```
<form id="search">
       <input type="search" placeholder="" name="search" value="<?= get('search') ?>"
         hx-get="/api/projects" hx-trigger="search" hx-target="#projects" hx-include="#filter" hx-push-url="/projects">
    </form>
```

It works, so on change I get /projects?search=xyz, but it won't include my parameters from the #filter form.
Ideally I want my two forms work together, but they are on different location in the html on the page…

**Comment by pbowyer** [comment]
Hi @fabianwohlfart I had to add `hx-push-url="true"` to make my form update the URL.
